src/uct/ib/ud/base/ud_inl.h

```diff
@@ -79,11 +79,42 @@ uct_ud_skb_release(uct_ud_send_skb_t *skb, int is_inline)
     }
 }
 
+#if UCS_ENABLE_ASSERT
+static UCS_F_ALWAYS_INLINE int uct_ud_ep_has_pending(uct_ud_ep_t *ep)
+{
+    return !ucs_arbiter_group_is_empty(&ep->tx.pending.group) &&
+           !ucs_arbiter_elem_is_only(&ep->tx.pending.elem);
+}
+#endif
+
+static UCS_F_ALWAYS_INLINE void uct_ud_ep_set_has_pending_flag(uct_ud_ep_t *ep)
+{
+    ep->flags |= UCT_UD_EP_FLAG_HAS_PENDING;
+}
+
+static UCS_F_ALWAYS_INLINE void uct_ud_ep_remove_has_pending_flag(uct_ud_ep_t *ep)
+{
+    ucs_assert(ep->flags & UCT_UD_EP_FLAG_HAS_PENDING);
+    ep->flags &= ~UCT_UD_EP_FLAG_HAS_PENDING;
+}
+
+static UCS_F_ALWAYS_INLINE void uct_ud_ep_set_dest_ep_id(uct_ud_ep_t *ep,
+                                                         uint32_t dest_id)
+{
+    ep->dest_ep_id = dest_id;
+
+    if (ep->dest_ep_id == UCT_UD_EP_NULL_ID) {
+        ep->flags &= ~UCT_UD_EP_FLAG_CONNECTED;
```

Comment:
i think that this function can be used only when dest_id!=UCT_UD_EP_NULL_ID (and revert UCT_UD_EP_NULL_ID usage to simple assign). this way no need to check, since it's just one case.

Reply:
done